pkg/controller/direct/workstations/config_defaults.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workstations
+
+import (
+	pb "cloud.google.com/go/workstations/apiv1/workstationspb"
+	krm "github.com/GoogleCloudPlatform/k8s-config-connector/apis/workstations/v1alpha1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct"
+)
+
+func ApplyWorkstationConfigGCPDefaults(mapCtx *direct.MapContext, in *krm.WorkstationConfigSpec, out *pb.WorkstationConfig, actual *pb.WorkstationConfig) {
```

Comment:
Will the Workstation service changes the default value or makes a difference between the fields that are unset (what user intend) or set to a default value (what KCC does).

Reply:
I'm ok with the current approach, ideally we want to separate the create and update handling because logically they are different. 
You may also want to double check the `ReplicaZones`, if it is set to empty slice (rather than unset), will the GCP service gives a default value, reject the call or unset the zones. (I recalled it uses the default value, which means we may want to deal `len(in.ReplicaZones) == 0` as well.

Reply:
It is impossible to update the `ReplicaZones` field (it's immutable).